File: src/model/LinkEntity.ts
import {Book} from './Book';
import {Client} from './Client';
import {Entity} from './Entity';

export class LinkEntity extends Entity{
  book: Book;
  client: Client;

  constructor(id?:number,book?:Book, client?:Client){
    super(id);
    this.book = book;
    this.client = client;
  }
}

File: catalog-core/src/main/java/ro/ubb/catalog/core/model/LinkEntity.java
package ro.ubb.catalog.core.model;

import lombok.*;
import ro.ubb.catalog.core.model.BaseEntity;
import ro.ubb.catalog.core.model.Book;
import ro.ubb.catalog.core.model.Client;

@NoArgsConstructor
@AllArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
@Builder
public class LinkEntity extends BaseEntity {
    Client client;
    Book book;

}
